### DISCUSSION TITLE
Update the project URL

### DISCUSSION THREAD

**Discussion by amithkoujalgi** [langchain4j/langchain4j#596] [Ideas]
Hello @langchain4j team!

I previously collaborated with @LizeRaes, and together we created a GitHub action to build and push docs to GitHub Pages: 
[Docs GitHub Action](https://github.com/langchain4j/langchain4j/blob/main/.github/workflows/docs.yaml)

I'm glad that this PR has been merged! 🤩 Thank you for that. 🙏🏽

However, I noticed that in the project settings, the URL is still pointing to the Langchain4j GitHub repository instead of the docs endpoint.

I was wondering if the project URL could be updated in the project settings to point to the docs endpoint. It would be really convenient to navigate to the [docs link](https://langchain4j.github.io/langchain4j/) directly from the GitHub repository.

**Comment by LizeRaes** [comment]
Hi @amithkoujalgi thanks a lot, please go ahead!
I had some other things on the docu backlog, in case you're on a roll:

- [x] We were thinking it would be even nicer to have the docu at https://langchain4j.github.io instead of the current https://langchain4j.github.io/langchain4j/ , I'm not sure if this is possible with the way Pages is set up, but feel free to change.

- [ ] If you find time, you're very welcome to publish the javadoc next to the docu on https://langchain4j.github.io/ (currently the javadoc is pointing to my pages (https://lizeraes.github.io/langchain4j/apidocs/)).

- [ ] The javadoc would need to be republished now the new release is out, it might be possible to include the modules that require higher java versions. These modules are currently excluded in the javadoc yaml, but it's worth a try to see if building it with java 21 would allow these modules to be included.
- [ ] Also the docu deserves a republish, I have .md changes merged that are not published yet. 

I don't expect to find time myself until next Wednesday, so the playground is all yours :)
Thank you very much in any case!

**Comment by amithkoujalgi** [comment]
Hello @LizeRaes,

Please find my responses inline.

> Hi @amithkoujalgi thanks a lot, please go ahead! 

This setting needs to be updated by the repo owner/admin 🙂

> * [ ]  We were thinking it would be even nicer to have the docu at https://langchain4j.github.io instead of the current https://langchain4j.github.io/langchain4j/ , I'm not sure if this is possible with the way Pages is set up, but feel free to change.

I updated the link in the documentation to make docs accessible at https://langchain4j.github.io instead of https://langchain4j.github.io/langchain4j/ and tried to get it to work but it did not work. When I looked up about this issue, I found that it's intentional that GitHub pages introduces the repository name as suffix. This is to differentiate between the GitHub pages deployed from different repositories of a GitHub account. For example, if I have `repo1` and `repo2` in my account and I have enabled GitHub pages on both these repositories, GitHub has to serve pages for both these repos and therefore it introduces the repository name suffix. So GitHub would enable URLs as `amithkoujalgi.github.io/repo1` and `amithkoujalgi.github.io/repo2`. So, this is more of a limitation if we use the domain provided by GitHub. However, I believe this could be resolved if a custom domain is mapped in the repository settings. I'm not 100% sure about this, but I could definitely give it a try if a custom domain could be setup in repo settings.

> * [ ]  If you find time, you're very welcome to publish the javadoc next to the docu on https://langchain4j.github.io/ (currently the javadoc is pointing to my pages (https://lizeraes.github.io/langchain4j/apidocs/)).

I was able to build and publish the javadoc alongside the docs directory. I've modified the docs GH workflow to autobuild javadoc along with the docs. However, I am using the following mvn command to build the javadoc which would build the javadoc under `target/site` directory.
```
mvn javadoc:aggregate -DskipTests
```
I found this command from another GH workflow and I am unsure if this is the latest command to build javadoc for this multi-module Maven project. Could you verify this?

> * [ ]  The javadoc would need to be republished now the new release is out, it might be possible to include the modules that require higher java versions. These modules are currently excluded in the javadoc yaml, but it's worth a try to see if building it with java 21 would allow these modules to be included.

I have set it up in the docs GH workflow. I have included all the modules in the javadoc build and I've switched to Java version 21. It seems to be building all the modules just file. Could you verify this?

> * [ ]  Also the docu deserves a republish, I have .md changes merged that are not published yet.
> 

Alright, I think the repo owner/admin needs to be notified to run the GH workflow for the docs to be republished.

Here's the PR I've created for the above mentioned changes: https://github.com/langchain4j/langchain4j/pull/599

**Reply by amithkoujalgi** [reply]
This URL remains to be updated to https://langchain4j.github.io/langchain4j 🙂 This needs to be done by repo owner/admin.

<img width="2039" alt="URL" src="https://github.com/langchain4j/langchain4j/assets/1876165/0ac596ea-63b3-4e5d-9b2f-412b8192efa2">
